src/Time/TimeSteppers/AdamsBashforthN.hpp

```diff
@@ -369,7 +375,33 @@ void AdamsBashforthN::update_u(
     const gsl::not_null<Vars*> u,
     const gsl::not_null<History<Vars, DerivVars>*> history,
     const TimeDelta& time_step) const noexcept {
-  update_u_impl(u, *history, time_step);
+  update_u_impl(u, *history, time_step, order_);
+  history->mark_unneeded(history->begin() + 1);
+}
+
+template <typename Vars, typename DerivVars>
+void AdamsBashforthN::update_u(
+    const gsl::not_null<Vars*> u,
+    const gsl::not_null<std::optional<Vars>*> u_error,
+    const gsl::not_null<History<Vars, DerivVars>*> history,
+    const TimeDelta& time_step) const noexcept {
+  if(order_ == 1) {
+    ERROR(
+        "Cannot reliably estimate the error for an order = 1 Adams-Bashforth "
+        "scheme. Consider using a higher order for the time-stepper.");
+  }
+  u_error->reset();
```

Comment:
Put this in an else branch for the next if.  No reason to throw away the passed object and then reallocate it.  (The assignment might cause a reallocation anyway, but no reason to force it to.)